### PR TITLE
Add ability to create, rename, and delete Service Types

### PR DESCRIPTION
**Pull request by martha**
## Description

PT issue: https://www.pivotaltracker.com/n/projects/2591838#

Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/4128

This PR adds UI elements around creation, deletion, and renaming of custom service types. We decided to leave custom service categories out of the UI for now, though the mutations are there for us to use later.

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/admin/components/services/NewServiceTypeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      setDialogOpen(false);
+    },
+  });
+
+  const {
+    data: pickListData,
+    loading: pickListLoading,
+    error: pickListError,
+  } = useGetPickListQuery({
+    variables: { pickListType: PickListType.AllServiceCategories },
```
Comment (martha) [inline src/modules/admin/components/services/NewServiceTypeButton.tsx:48]: I was wondering if we will need to create a new pick list for only non-HUD service categories?

Comment (gigxz) [reply src/modules/admin/components/services/NewServiceTypeButton.tsx:48]: Hmm good question. I think this is okay, afaik there is no logical reason why custom services couldn't be inside a group with other hud services.

File: src/modules/admin/components/services/NewServiceTypeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+const NewServiceTypeButton = () => {
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [name, setName] = useState('');
+  const [supportsBulkAssignment, setSupportsBulkAssignment] = useState(false);
+  const [serviceCategory, setServiceCategory] = useState<PickListOption | null>(
+    null
+  );
+
+  const [createServiceType, { error, loading }] = useMutation<
```
Comment (gigxz) [inline src/modules/admin/components/services/NewServiceTypeButton.tsx]: You should be able to use `useCreateServiceTypeMutation` here. It is generated into `gqlTypes.ts`

File: src/modules/admin/components/services/ServiceTypeDetailPage.tsx [hunk 3]
```diff
@@ -20,30 +45,83 @@ const ServiceTypeDetailPage = () => {
     variables: { id: serviceTypeId },
   });
 
+  const navigate = useNavigate();
+
+  const [editDialogOpen, setEditDialogOpen] = useState(false);
+  const [editedName, setEditedName] = useState<string>();
+
+  const [renameServiceType, { error: renameError, loading: renameLoading }] =
+    useMutation<RenameServiceTypeMutation, RenameServiceTypeMutationVariables>(
```
Comment (gigxz) [inline src/modules/admin/components/services/ServiceTypeDetailPage.tsx]: Same here on using generated hook useRenameServiceTypeMutation

**Review by gigxz — APPROVED** [review]
looks good!

**Pull request merged**